tests/installation/grub_utils.pm

```diff
@@ -0,0 +1,80 @@
+package grub_utils;
+
+use base Exporter;
+use Exporter;
+
+use strict;
+use warnings;
+use opensusebasetest qw(handle_uefi_boot_disk_workaround);
+use testapi;
+use utils;
+use version_utils qw(is_sle is_livecd);
+use bootloader_setup qw(stop_grub_timeout boot_into_snapshot);
+
+our @EXPORT = qw(
+  grub_test
+);
+
+=head2 handle_installer_medium_bootup
+
+Due to pre-installation setup, qemu boot order is always booting from CD-ROM.
+=cut
+sub handle_installer_medium_bootup {
+    return unless (check_var("BOOTFROM", "d") || (get_var('UEFI') && get_var('USBBOOT')));
+    assert_screen 'inst-bootmenu', 180;
+
+    if (check_var("BOOTFROM", "d") && check_var("AUTOUPGRADE") && check_var("PATCH")) {
+        assert_screen 'grub2';
+    }
```

Comment:
There is no action between `assert-screen 'inst-bootmenu'` and `assert_screen 'grub2'`. If this is supposed to just look at the same screen a second time I suggest to decide about the correct tag to test in the first place, e.g.:

```suggestion
    assert_screen (check_var("BOOTFROM", "d") && check_var("AUTOUPGRADE") && check_var("PATCH"))) ? 'grub2' : 'inst-bootmenu';
```

Reply:
I just copied all those functions from `grub_test` test module and for me this condition does not make sense at all, because `check_var("AUTOUPGRADE")` and `check_var("PATCH")` are used here without second parameter. 

Do you have an idea of what should be there? For me this looks like some part of code that never worked.

Reply:
Oh, I see now. Well, I assume this has to do with upgrade test cases and also I assume despite being not completely correct it still does what the author expected. Maybe `check_var('FOO')` is effectively just the same as `get_var('FOO')`. However, if you copy-pasted that from elsewhere that is already bad and should be avoided. I see no way around by extracting functions for the common code and reduce the duplication as much as possible or as alternative completely replace all code that you consider legacy.

Reply:
IMO those `check_var()` always return false as `testapi::check_var()` does this:
```
    return 1 if (defined $bmwqemu::vars{$var} && $bmwqemu::vars{$var} eq $val);
    return 0;
```